### PR TITLE
Add option --hex-dump to dump out bytes stored in individual CAS Blocks

### PR DESCRIPTION
**Pull request by rastogishubham**

### CODE REVIEW

**Comment by rastogishubham** [comment]
@swift-ci please test

File: llvm/tools/llvm-cas-dump/MCCASPrinter.cpp [hunk 1]
```diff
@@ -78,6 +78,21 @@ Error MCCASPrinter::printMCObject(MCObjectProxy MCObj, DWARFContext *DWARFCtx) {
   OS.indent(Indent);
   OS << formatv("{0, -15} {1} \n", MCObj.getKindString(),
                 MCSchema.CAS.getID(MCObj));
+  if (Options.HexDump) {
```
Comment (felipepiovezan) [inline llvm/tools/llvm-cas-dump/MCCASPrinter.cpp:81]: LLVM has a lot of functional-style helpers to do this with less state:


```cpp
while (!data.empty()) {
   OS.indent(Indent);
   llvm::interleave(data.take_front(8), [](char c) {return llvm::format_hex(c, 4);}, " ");
   data = data.drop_front(8);
}
```

**Review by felipepiovezan — COMMENTED** [review]
Could we also add a test for this? We can have an empty module (`.ll`) but edit the compiler string to give it some easily recognizable name like: "ABCDEFG" and then find that in the dump of the CAS block for the string section

**Comment by rastogishubham** [comment]
@swift-ci please test

**Comment by rastogishubham** [comment]
@swift-ci please test

**Review by felipepiovezan — APPROVED** [review]
LGTM!
